cmd/vela-img/build.go

```diff
@@ -4,15 +4,239 @@
 
 package main
 
-import "github.com/sirupsen/logrus"
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+const buildAction = "build"
 
 // Build represents the plugin configuration for build information.
 type Build struct {
+	// BuildArg should set build time variables
+	BuildArgs []string
+	// CacheFrom should be images to consider as cache sources
+	CacheFrom []string
+	// directory should be a path to the context you want docker to run
+	Directory string
+	// File should be name and path to the Dockerfile
+	File string
+	// Labels should be set metadata for an image
+	Labels []string
+	// NoCache should be do not use cache when building the image
+	NoCache bool
+	// NoConole should be non-console progress UI
+	NoConsole bool
+	// Output BuildKit output specification (e.g. type=tar,dest=build.tar)
+	Output string
+	// Platform should be platforms for which the image should be built
+	Platforms []string
+	// Tag should be name and optionally a tag in the 'name:tag' format
+	Tags []string
+	// Target should be the target build stage to build
+	Target string
+}
+
+// buildFlags represents for config settings on the cli.
+var buildFlags = []cli.Flag{
+	&cli.StringSliceFlag{
+		Name:     "build.build-args",
+		Usage:    "should be images to consider as cache sources",
+		EnvVars:  []string{"PARAMETER_BUILD_ARGS", "BUILD_BUILD_ARGS"},
+		FilePath: string("/vela/parameters/img/build/build_args,/vela/secrets/img/build/build_args"),
+	},
+	&cli.StringSliceFlag{
+		Name:     "build.cache-from",
+		Usage:    "should set build time variables",
+		EnvVars:  []string{"PARAMETER_CACHE_FROM", "BUILD_CACHE_FROM"},
+		FilePath: string("/vela/parameters/img/build/cache_from,/vela/secrets/img/build/cache_from"),
+	},
+	&cli.StringFlag{
+		Name:     "build.directory",
+		Usage:    "should be a path to the context you want docker to run",
```

Comment:
i approved but if you feel like it, this instance of `docker` could be `img`? :D

Reply:
updated